src/app-layout/split-panel/provider.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useState } from 'react';
+import { getLimitedValue } from '../../split-panel/utils/size-utils';
+import { SplitPanelContextBaseProps, SplitPanelContextProvider } from '../../internal/context/split-panel-context';
+
+export interface SplitPanelProviderProps extends SplitPanelContextBaseProps {
+  getMaxWidth: () => number;
+  getMaxHeight: () => number;
+  children?: React.ReactNode;
+}
+
+const MIN_HEIGHT = 160;
+const MIN_WIDTH = 280;
+
+export function SplitPanelProvider({
```

Comment:
Why a separate provider is necessary, can the SplitPanelContextProvider be modified instead?

Reply:
This is not a modifiable component: https://github.com/cloudscape-design/components/blob/b0e7f9373943955df01221aead7b0ce26a544ba3/src/internal/context/split-panel-context.ts#L38-L40